CMakeLists.txt

```diff
@@ -76,6 +76,11 @@ set(FlatBuffers_Sample_Text_SRCS
 # source_group(Compiler FILES ${FlatBuffers_Compiler_SRCS})
 # source_group(Tests FILES ${FlatBuffers_Tests_SRCS})
 
+# Certain platforms such as ARM do not use signed chars by default
+# which causes issues with certain bounds checks.
+set(CMAKE_CXX_FLAGS
+  "${CMAKE_CXX_FLAGS} -fsigned-char")
+
```

Comment:
Setting this globally may cause problems with compilers that are not gcc/clang, such as visual studio. I would just add this flag to the gcc/clang lines below (I'll refactor these flags someday).